New Resource: `azurerm_certificate_order_key_vault_store` (#25464)
terraform-providers/terraform-provider-azurerm: website/docs/r/app_service_certificate_order_certificate.html.markdown

      "List"
    ]
  }

  // Microsoft.Azure.CertificateRegistration object ID
  access_policy {
    tenant_id = data.azurerm_client_config.test.tenant_id
    object_id = "ed47c2a1-bd23-4341-b39c-f4fd69138dd3"

    secret_permissions = [
      "Delete",
      "Get",
      "Purge",
      "Set",
      "List"
    ]

    certificate_permissions = [
      "Create",
      "Delete",
      "Get",
      "Purge",
      "Import",
      "List"
    ]
  }
}

resource "azurerm_app_service_certificate_order" "example" {
  name                = "example-cert-order"
  resource_group_name = azurerm_resource_group.example.name
  location            = "global"
  distinguished_name  = "CN=example.com"
  product_type        = "Standard"
}

resource "azurerm_app_service_certificate_order_certificate" "test" {
  name                  = "example-certorder-cert"
  certificate_order_id  = azurerm_app_service_certificate_order.example.id
  key_vault_id          = azurerm_key_vault.example.id
  key_vault_secret_name = "example-keyvault-secret"
}
```

-> **Note:** Please make sure the domain ownership is verified before configure the key vault.

## Argument Reference

The following arguments are supported:

> * `name` - (Required) Specifies the name of the certificate order certificate key vault binding. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `name` - (Required) Specifies the name of the Certificate Key Vault Store. Changing this forces a new resource to be created.
```

> * `certificate_order_id` - (Required) The id of the certificate order in which to configure the certificate. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `certificate_order_id` - (Required) The ID of the Certificate Order in which to configure the Certificate Key Vault Store Binding. Changing this forces a new resource to be created.
```

> * `key_vault_id` - (Required) The id of the key vault in which to bind the certificate order certificate.

Reviewer: ```suggestion
* `key_vault_id` - (Required) The ID of the Key Vault in which to bind the Certificate.
```

> * `key_vault_secret_name` - (Required) The name of the key vault secrete in which to bind the certificate order certificate.

Reviewer: ```suggestion
* `key_vault_secret_name` - (Required) The name of the Key Vault Secret to bind to the Certificate.
```

## Attributes Reference

> * `location` - The location of the certificate order certificate.

Reviewer: ```suggestion
* `location` - The location of the Certificate.
```

> * `type` - The type of the certificate order certificate.

Reviewer: ```suggestion
* `type` - The type of Certificate.
```
